content/documents/vercel-integration.mdx

```diff
@@ -58,6 +58,13 @@ The OrderCloud integration is part of the Vercel integration marketplace. It can
 }
 ```
 
+Why two API clients to connect to OrderCloud? The OrderCloud API offers extreme flexibility in commerce use cases. The trade-off is that custom solutions must bring their own definitions of the various user roles and permissions they want to support. OrderCloud provides tools for controlling which users can take what API actions and which users can authenticate through which API clients. However, making those configuration choices and then securely connecting your custom code are a complexity that falls on the solution implementor. With this Vercel integration, you get a pre-coded storefront and a pre-configured data template. We've made the user roles and security decisions for you by choosing a very common pattern. 
+
+The `Vercel-Storefront-Connector` client is for buyer users. It is used to authenticate requests to ordercloud that render the storefront like the PLP, PDP, Cart, Checkout, etc. It is given limitted API permissions - for example, you can't use it to change product features or read orders accross users. These would be inappropriate things for a buyer to do. Because its permissions are limitted, it is safe to expose the details to a client browser. It is the safer, more common method, being used to make the vast majority of requests to OrderCloud. The `Vercel-Middleware-Connector` client, on the other hand, is given full API access. It can read or modify any piecce of data in the OrderCloud marketplace. Accordingly, it also requires a Client Secret which should never be public. It is only used inside serverless functions which do pre-canned actions after appropriate security checks. In this app it is used in only 1 place, to set Payment.Accepted to true after user's credit card details are considered captured. 
+
+Hopefully this explains why 2 API Clients are needed by this integration.   
```

Comment:
Could this be spun in a more concrete way?  Maybe instead of "Hopefully this explains ..." it could be some info on where to seek help if the reader has further questions?  I think we should assume the article _does_ explain why 2 API Clients are needed.

Reply:
updated!